.github/workflows/update-pre-release-branches.yaml

```diff
@@ -0,0 +1,77 @@
+name: Auto-update pre-release branches
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - .github/workflows/update-pre-release-branches.yaml
+  schedule:
+    - cron: "0 0 * * *" # Runs every midnight
+
+permissions:
+  contents: read
+
+jobs:
+  update-branches:
+    permissions:
+      contents: write  # for Git to git push
+    runs-on: ubuntu-latest
+    outputs:
+      preRelease: ${{ steps.determine.outputs.preRelease }}
+      branch: ${{ steps.determine.outputs.branch }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ssh-key: ${{ secrets.BOT_SSH_KEY }}
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Install Python dependencies
+        run: pip3 install packaging requests
+      - name: Determine outstanding pre-release
+        id: determine
+        run: |
+          preRelease=`python3 ./build-scripts/k8s_releases.py get_outstanding_prerelease`
+          echo "preRelease=$preRelease" >> "$GITHUB_OUTPUT"
+
+          if [[ -n "$preRelease" ]]; then
+            branch="autoupdate/$preRelease"
+          fi
+          echo "branch=$branch" >> "$GITHUB_OUTPUT"
+      - name: Define git credentials
+        run: |
+          # Needed to create commits.
+          git config --global user.name "Github Actions"
+          git config --global user.email "worker@org.com"
+      - name: 'Update k8s component version: ${{ steps.determine.outputs.preRelease }}'
+        if: ${{ steps.determine.outputs.preRelease }} != ''
+        run: |
+          echo ${{ steps.determine.outputs.preRelease }} > ./build-scripts/components/kubernetes/version
+          git add ./build-scripts/components/kubernetes/version
+          git commit -m "Update k8s version to ${{ steps.determine.outputs.preRelease }}"
```

Comment:
please move all of those steps into a script, similar to https://github.com/canonical/k8s-snap/blob/30d3f3b8429fb84e3df94a32928fcb1605403671/build-scripts/hack/update-component-versions.py#L1